### PR TITLE
[SW-1247] feat: improve invite links

### PR DESCRIPTION
**Pull request by CedrikNikita**

### CODE REVIEW

**Comment by Liubov-crypto** [comment]
LGTM

**Comment by Liubov-crypto** [comment]
LGTM

File: src/popup/components/AccountSelector.vue [hunk 1]
```diff
@@ -108,5 +109,9 @@ export default defineComponent({
       margin-bottom: 0;
     }
   }
+
+  .address-truncated {
```
Comment (peronczyk) [inline src/popup/components/AccountSelector.vue:113]: There is no such class in the template.

File: src/popup/components/AddressTruncated.vue [hunk 2]
```diff
@@ -32,15 +37,18 @@ import { PROTOCOL_AETERNITY } from '@/constants';
...
 
 export default defineComponent({
   components: {
+    ProtocolIcon,
     LinkButton,
     ExternalLinkIcon,
   },
   props: {
     address: { type: String, required: true },
+    showProtocolIcon: Boolean,
```
Comment (peronczyk) [inline src/popup/components/AddressTruncated.vue:51]: Please move this line below the `protocol`. The reason is that it's a good pattern to group Boolean props together. Also it makes more sense in terms of alphabetical order.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+    class="claim-gift-card"
+    @close="reject"
+  >
+    <div class="header text-heading-2">
+      {{ $t('modals.claimGiftCard.title') }}
+    </div>
+    <template v-if="step === STEPS.redeemPartially || (step === STEPS.redeemFull && !isCardEmpty)">
+      <CheckCircleIcon class="check-circle-icon" />
+      <div class="redeemed-amount">
+        <span>{{ `${amount} AE` }}</span>
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:15]:
```suggestion
        <span>{{ amount }} {{ AE_SYMBOL }}</span>
```

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+        />
+      </div>
+    </template>
+  </Modal>
+</template>
+
+<script lang="ts">
+import BigNumber from 'bignumber.js';
+import {
+  computed, defineComponent, ref, PropType, onMounted,
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:97]: I suggest to split this line.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+import {
+  useAccounts,
+  useAeSdk,
+  useCurrencies,
+} from '@/composables';
+import { AE_COIN_PRECISION } from '@/protocols/aeternity/config';
+import { getAccountFromSecret } from '@/protocols/aeternity/helpers';
+import { PROTOCOL_AETERNITY } from '@/constants';
+import CheckCircleIcon from '@/icons/check-circle.svg?vue-component';
+import type { RejectCallback, ResolveCallback, ObjectValues } from '../../../types';
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:121]: I suggest to put the `types` import above all local imports and use the `@` in the import path.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+    align-self: center;
+    margin: 8px auto 4px auto;
+  }
+
+  .redeemed-amount {
+    display: flex;
+    justify-content: center;
+    color: variables.$color-white;
+
+    @extend %face-sans-16-medium;
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:321]: Please put the extend at the beginning of the definitions for this class.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 7]
```diff
@@ -0,0 +1,406 @@
...
+
+    line-height: 24px;
+
+    .gray {
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:325]: Name "gray" has nothing to do with the opacity and margin.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 8]
```diff
@@ -0,0 +1,406 @@
...
+      margin-left: 4px;
+    }
+  }
+
+  .details-item {
+    display: flex;
+    justify-content: center;
+
+    .account-selector {
+      :deep(.truncate) {
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:336]: Parent components should not affect child component internals. So please avoid using `:deep` selector. If you want to make a child component to behave or look differently please use bool prop and update the child component itself.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 9]
```diff
@@ -0,0 +1,406 @@
...
+.claim-gift-card {
+  :deep(.container) {
+    min-height: 480px;
+  }
+
+  .header {
+    text-align: center;
+    margin-bottom: 4px;
+
+    @extend %face-sans-19-medium;
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:304]: Please put the extend at the top of the class definitions.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 10]
```diff
@@ -0,0 +1,406 @@
...
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+@use '../../../styles/variables';
+@use '../../../styles/typography';
+
+.claim-gift-card {
+  :deep(.container) {
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:296]: Parent components should not affect child component internals. So please avoid using :deep selector. If you want to make a child component to behave or look differently please use bool prop and update the child component itself.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 11]
```diff
@@ -0,0 +1,406 @@
...
+      STEPS,
+      updateBalance,
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+@use '../../../styles/variables';
+@use '../../../styles/typography';
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:293]: You can use `@` in the paths.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 12]
```diff
@@ -0,0 +1,406 @@
...
+          variant="muted"
+          :text="$t('modals.claimGiftCard.redeemPartially')"
+          :disabled="loading"
+          @click="step = STEPS.form"
+        />
+        <BtnMain
+          :icon="loading ? AnimatedSpinner : null"
+          :text="mainButtonText"
+          :disabled="isDisabled || loading"
+          @click="handleMainButtonClick"
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:87]:
```suggestion
          @click="handleMainButtonClick()"
```
This avoids passing the click event as the 1st function argument.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 13]
```diff
@@ -0,0 +1,406 @@
...
+        max_redeem: max.toString(),
+      }"
+    >
+      <template #label-after>
+        <BtnPlain
+          class="max-button"
+          :class="{ chosen: amount.toString() === max.toString() }"
+          @click="setMaxAmount"
+        >
+          {{ $t('common.max') }}
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:70]: I suggest to use `text` prop to make the code slimmer.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 14]
```diff
@@ -0,0 +1,406 @@
...
+    const fee = BigNumber(unpackTx(
+        buildTx({
+          tag: Tag.SpendTx,
+          senderId: address,
+          recipientId: recipientId.value,
+          amount: +amount.value,
+          payload: encode(new TextEncoder().encode(''), Encoding.Bytearray),
+          nonce: 1,
+        }) as any,
+        Tag.SpendTx, // https://github.com/aeternity/aepp-sdk-js/issues/1852
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:183]: Wrong indentation.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 15]
```diff
@@ -0,0 +1,406 @@
...
+    const isDisabled = computed(() => {
+      switch (step.value) {
+        case STEPS.form:
+          return !amount.value || !!errors.value?.amount;
+        default:
+          return false;
+      }
+    });
+
+    const currencyFormatted = computed(() => getFormattedFiat(+amount.value));
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:255]: I suggest to move the computed properties before the function definitions.

File: src/popup/components/Modals/ClaimGiftCard.vue [hunk 16]
```diff
@@ -0,0 +1,406 @@
...
+    text-align: center;
+    margin-bottom: 4px;
+
+    @extend %face-sans-19-medium;
+  }
+
+  .check-circle-icon {
+    width: 40px;
+    height: 40px;
+    color: #00d3a1;
```
Comment (peronczyk) [inline src/popup/components/Modals/ClaimGiftCard.vue:310]: Please use colors saved in the variables file. If it's a new color add it to mentioned file.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/store/modules/invites.js [hunk 17]
```diff
@@ -18,25 +15,30 @@ export default {
     },
   },
   actions: {
-    async claim({ rootGetters, rootState }, secretKey) {
+    async claim(_, {
+      secretKey, recipientId, amount = undefined, isMax,
```
Comment (peronczyk) [inline src/store/modules/invites.js:19]: Please split this line.

File: src/protocols/aeternity/helpers/index.ts [hunk 18]
```diff
@@ -348,3 +350,11 @@ export function checkAddress(value: string) {
 export function checkAddressOrChannel(value: string) {
   return checkAddress(value) || isAddressValid(value, Encoding.Channel);
 }
+
+const SEED_LENGTH = 32;
```
Comment (peronczyk) [inline src/protocols/aeternity/helpers/index.ts:354]: I suggest to move this into global constants file as this is not a "helper" and we use the same seed for all the protocols.

File: src/protocols/aeternity/helpers/index.ts [hunk 19]
```diff
@@ -348,3 +350,11 @@ export function checkAddress(value: string) {
...
+
+export function getAccountFromSecret(secretKey: Buffer) {
+  // `secretKey` variable can be either seed or seed + public key (legacy)
+  return new MemoryAccount(Buffer.from(secretKey).length === SEED_LENGTH
+    ? nacl.sign.keyPair.fromSeed(Buffer.from(secretKey)).secretKey : Buffer.from(secretKey));
```
Comment (peronczyk) [inline src/protocols/aeternity/helpers/index.ts:359]:
```suggestion
  return new MemoryAccount(
    Buffer.from(secretKey).length === SEED_LENGTH
      ? nacl.sign.keyPair.fromSeed(Buffer.from(secretKey)).secretKey
      : Buffer.from(secretKey),
  );
```
Please make it more readable by splitting the line.

**Review by peronczyk — CHANGES_REQUESTED** [review]

**Review by peronczyk — APPROVED** [review]

File: src/store/modules/invites.js [hunk 20]
```diff
@@ -21,25 +18,33 @@ export default {
...
-      );
+      if (!isMax) {
+        await aeSdk.spend(
+          amount,
+          recipientId,
+          { denomination: AE_AMOUNT_FORMATS.AE },
+        );
+      } else {
+        await aeSdk.transferFunds(
+          1,
```
Comment (pstachel) [inline src/store/modules/invites.js:43]: I think we should avoid magic numbers

**Review by pstachel — APPROVED** [review]

**Comment by Liubov-crypto** [comment]
LGTM

**Pull request merged**